spring-boot-project/spring-boot/src/main/java/org/springframework/boot/logging/logback/SpringPropertyModel.java

```diff
@@ -58,4 +58,11 @@ void setSource(String source) {
 		this.source = source;
 	}
 
+	String getSourceOrDefaultValue() {
+		if (this.source == null) {
+			return this.defaultValue;
+		}
+		return this.source;
+	}
```

Comment:
I'm not certain if it's appropriate to find the default values here.
However, since the logic was handled within the object, I made sure this class took responsibility for it.